xbmc/guilib/GUIControl.cpp

```diff
@@ -472,8 +482,11 @@ float CGUIControl::GetHeight() const
   return m_height;
 }
 
-void CGUIControl::MarkDirtyRegion(const unsigned int dirtyState)
+void CGUIControl::MarkDirtyRegion(const unsigned int dirtyState, bool checkTransparency)
```

Comment:
is the new argument and the bail really required? I'd say that if you set the control dirty on process (after deciding on the transparency) you simply mark it as dirty and go with it

Reply:
The bail is required, but the argument is not.

If the element is invisible, we don't want a dirty region to appear unnecessarily. If it comes to life after the animation, it still can be marked as dirty.

If a state change occurs from opaque to transparent, it should not set to be dirty (it was a bit late when I wrote this). And there lies a tiny flaw with this implementation, as it would still happen regardless. The surface might be marked as dirty before the animation says it is not visible. But this just creates a one frame "lag" in some special cases, which is not critical in my eyes.